src/kvtree_util.c

```diff
@@ -225,3 +230,130 @@ int kvtree_util_get_ptr(const kvtree* hash, const char* key, void** val)
 
   return rc;
 }
+
+/*
+ * Given a prefix passed to a kvtree_write_gather() (like "/tmp/rank2file",
+ * read all the rank data from all the scattered subfiles, and construct a
+ * kvtree in 'data'.  'data' will look something like:
+ *
+ *       26
+ *         FILE
+ *           ckpt.1/rank_26.ckpt
+ *       24
+ *         FILE
+ *           ckpt.1/rank_24.ckpt
+ *       25
+ *         FILE
+ *           ckpt.1/rank_25.ckpt
+ *       18
+ *         FILE
+ *           ckpt.1/rank_18.ckpt
+ *       ...
+ *
+ *  ... with each top-level element being the rank number.
+ */
+int kvtree_read_scatter_single(const char* prefix, kvtree* data)
```

Comment:
Oh, I should have found this earlier.  Let's move ``kvtree_read_scatter_single`` (and its new includes) to kvtree.c just after ``kvtree_write_to_gather``.  It's effectively the read counterpart to that write function.